src/components/people.tsx

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable max-len */
+import React, { useContext, useEffect, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { getPeople } from '../api';
+import { MyContext } from './state';
+import { Loader } from './Loader';
+import { Error } from './error';
+import { Person } from './person';
+import { Person as PersonType } from '../types';
+import { PeopleFilters } from './PeopleFilters';
+
+export const People = () => {
+  const { person, setPerson, error, setError, setIsLoading, isLoading } =
+    useContext(MyContext);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [originalPeople, setOriginalPeople] = useState<PersonType[]>([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(fetchedPeople => {
+        setPerson(fetchedPeople);
+        setOriginalPeople(fetchedPeople);
+      })
+      .catch(() => {
+        setError('Unable to load people: ');
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const sortPeopleByField = (
+    field: keyof PersonType,
+    event: React.MouseEvent,
+  ) => {
+    event.preventDefault();
+
+    const sort = searchParams.get('sort');
+    const order = searchParams.get('order');
+
+    if (!sort || sort !== field) {
+      const sortedPeople = [...person].sort((a, b) =>
+        String(a[field]).localeCompare(String(b[field])),
+      );
+
+      setPerson(sortedPeople);
+      setSearchParams({ sort: field });
+    } else if (sort === field && !order) {
+      const sortedPeople = [...person].sort((a, b) =>
+        String(b[field]).localeCompare(String(a[field])),
+      );
+
+      setPerson(sortedPeople);
+      setSearchParams({ sort: field, order: 'desc' });
+    } else {
+      setPerson(originalPeople);
+      setSearchParams({});
+    }
+  };
+
+  const getSortIcon = (field: string) => {
+    const sort = searchParams.get('sort');
+    const order = searchParams.get('order');
+
+    if (!sort || sort !== field) {
+      return 'fas fa-sort';
+    } else if (sort === field && !order) {
+      return 'fas fa-sort-up';
+    } else if (sort === field && order === 'desc') {
+      return 'fas fa-sort-down';
+    }
+
+    return 'fas fa-sort';
+  };
```

Comment:
Similar to the `sortPeopleByField` function, the `getSortIcon` function can also be simplified and broken down into smaller functions. This would improve the readability and maintainability of your code.